File: src/StorePOS.Api.Integration.Guide/examples/javascript/storepos-client.js
class StorePosApiClient {
    constructor(baseUrl = 'http://localhost:5062') {
        this.baseUrl = baseUrl;
        this.accessToken = null;
    }
    
    async login(usernameOrEmail, password) {
        try {
            const response = await fetch(`${this.baseUrl}/api/auth/login`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    usernameOrEmail,
                    password
                })
            });
            
            const result = await response.json();
            
            if (result.success) {
                this.accessToken = result.data.accessToken;
                console.log('Login successful!');
                return true;
            } else {
                console.error('Login failed:', result.message);
                return false;
            }
        } catch (error) {
            console.error('Login error:', error);
            return false;
        }
    }
    
    async getProducts() {
        if (!this.accessToken) {
            throw new Error('Not authenticated. Please login first.');
        }
        
        try {
            const response = await fetch(`${this.baseUrl}/api/products`, {
                headers: {
                    'Authorization': `Bearer ${this.accessToken}`,
                    'Content-Type': 'application/json'
                }
            });
            
            if (response.ok) {
                return await response.json();
            } else {
                throw new Error(`HTTP ${response.status}: ${response.statusText}`);
            }
        } catch (error) {
            console.error('Error fetching products:', error);
            throw error;
        }
    }
    
    async createProduct(productData) {
        if (!this.accessToken) {
            throw new Error('Not authenticated. Please login first.');
        }
        
        try {
            const response = await fetch(`${this.baseUrl}/api/products`, {
                method: 'POST',
                headers: {
                    'Authorization': `Bearer ${this.accessToken}`,
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify(productData)
            });
            
            if (response.ok) {
                return await response.json();
            } else {
                throw new Error(`HTTP ${response.status}: ${response.statusText}`);
            }
        } catch (error) {
            console.error('Error creating product:', error);
            throw error;
        }
    }
    
    async updateProduct(id, productData) {
        if (!this.accessToken) {
            throw new Error('Not authenticated. Please login first.');
        }
        
        try {
            const response = await fetch(`${this.baseUrl}/api/products/${id}`, {
                method: 'PUT',
                headers: {
                    'Authorization': `Bearer ${this.accessToken}`,
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify(productData)
            });
            
            if (response.ok) {
                return await response.json();
            } else {
                throw new Error(`HTTP ${response.status}: ${response.statusText}`);
            }
        } catch (error) {
            console.error('Error updating product:', error);
            throw error;
        }
    }
    
    async deleteProduct(id) {
        if (!this.accessToken) {
            throw new Error('Not authenticated. Please login first.');
        }
        
        try {
            const response = await fetch(`${this.baseUrl}/api/products/${id}`, {
                method: 'DELETE',
                headers: {
                    'Authorization': `Bearer ${this.accessToken}`,
                    'Content-Type': 'application/json'
                }
            });
            
            return response.ok;
        } catch (error) {
            console.error('Error deleting product:', error);
            return false;
        }
    }
}

// Usage example for Node.js
async function example() {
    const client = new StorePosApiClient();
    
    if (await client.login('admin@storepos.com', 'Admin123!')) {
        try {
            // Get all products
            const products = await client.getProducts();
            console.log(`Retrieved ${products.length} products`);
            
            // Create a new product
            const newProduct = {
                sku: 'JS-001',
                barcode: '1234567890123',
                name: 'Product from JavaScript',
                category: 'Test',
                price: 25.99,
                cost: 15.00,
                stockQty: 100,
                isActive: true
            };
            
            const created = await client.createProduct(newProduct);
            console.log(`Created product with ID: ${created.id}`);
            
            // Update the product
            const updated = await client.updateProduct(created.id, {
                ...newProduct,
                name: 'Updated Product from JavaScript',
                price: 29.99
            });
            console.log(`Updated product: ${updated.name}`);
            
            // Delete the product
            const deleted = await client.deleteProduct(created.id);
            console.log(`Product deleted: ${deleted}`);
            
        } catch (error) {
            console.error('Failed to manage products:', error);
        }
    }
}

// Export for Node.js
if (typeof module !== 'undefined' && module.exports) {
    module.exports = StorePosApiClient;
}

// Run example if this is the main file
if (typeof require !== 'undefined' && require.main === module) {
    example();
}

File: src/StorePOS.Api.Integration.Guide/examples/python/storepos_client.py
import requests
import json
from typing import Optional, List, Dict

class StorePosApiClient:
    def __init__(self, base_url: str = "http://localhost:5062"):
        self.base_url = base_url
        self.access_token: Optional[str] = None
        self.session = requests.Session()
    
    def login(self, username_or_email: str, password: str) -> bool:
        """Login and store access token"""
        login_data = {
            "usernameOrEmail": username_or_email,
            "password": password
        }
        
        try:
            response = self.session.post(
                f"{self.base_url}/api/auth/login",
                json=login_data,
                headers={"Content-Type": "application/json"}
            )
            
            if response.status_code == 200:
                result = response.json()
                if result.get("success"):
                    self.access_token = result["data"]["accessToken"]
                    # Set default authorization header
                    self.session.headers.update({
                        "Authorization": f"Bearer {self.access_token}"
                    })
                    print("Login successful!")
                    return True
                else:
                    print(f"Login failed: {result.get('message')}")
                    return False
            else:
                print(f"HTTP Error {response.status_code}: {response.text}")
                return False
                
        except requests.exceptions.RequestException as e:
            print(f"Request error: {e}")
            return False
    
    def get_products(self) -> Optional[List[Dict]]:
        """Get all products"""
        if not self.access_token:
            raise Exception("Not authenticated. Please login first.")
        
        try:
            response = self.session.get(f"{self.base_url}/api/products")
            
            if response.status_code == 200:
                return response.json()
            else:
                print(f"Error getting products: {response.status_code}")
                return None
                
        except requests.exceptions.RequestException as e:
            print(f"Request error: {e}")
            return None
    
    def create_product(self, product_data: Dict) -> Optional[Dict]:
        """Create a new product"""
        if not self.access_token:
            raise Exception("Not authenticated. Please login first.")
        
        try:
            response = self.session.post(
                f"{self.base_url}/api/products",
                json=product_data
            )
            
            if response.status_code == 201:
                return response.json()
            else:
                print(f"Error creating product: {response.status_code}")
                return None
                
        except requests.exceptions.RequestException as e:
            print(f"Request error: {e}")
            return None

# Usage example
if __name__ == "__main__":
    client = StorePosApiClient()
    
    if client.login("admin@storepos.com", "Admin123!"):
        products = client.get_products()
        if products:
            print(f"Retrieved {len(products)} products")
            
            # Create a new product example
            new_product = {
                "sku": "PYTHON-001",
                "barcode": "1234567890123",
                "name": "Product from Python",
                "category": "Test",
                "price": 19.99,
                "cost": 12.00,
                "stockQty": 50,
                "isActive": True
            }
            
            created = client.create_product(new_product)
            if created:
                print(f"Created product with ID: {created['id']}")
        else:
            print("Failed to retrieve products")
